SharedPackages/BrowserServicesKit/Sources/PrivacyStats/internal/PrivacyStatsUtils.swift

```diff
@@ -98,6 +98,34 @@ final class PrivacyStatsUtils {
         return groupedResults
     }
 
+    /**
+     * Returns total count of blocked trackers across all companies for past 7 days.
+     */
+    static func load7DaysTotalBlockedTrackersCount(in context: NSManagedObjectContext) throws -> Int64 {
+        let startDate = Date().privacyStatsOldestPackTimestamp
+        return try loadTotalBlockedTrackersCount(since: startDate, in: context)
+    }
+
+    private static func loadTotalBlockedTrackersCount(since startDate: Date, in context: NSManagedObjectContext) throws -> Int64 {
+        let request = NSFetchRequest<NSFetchRequestResult>(entityName: DailyBlockedTrackersEntity.Const.entityName)
+        request.predicate = NSPredicate(format: "%K >= %@", #keyPath(DailyBlockedTrackersEntity.timestamp), startDate as NSDate)
+
+        // Expression description for the sum of count
+        let countExpression = NSExpression(forKeyPath: #keyPath(DailyBlockedTrackersEntity.count))
+        let sumExpression = NSExpression(forFunction: "sum:", arguments: [countExpression])
+
+        let sumExpressionDescription = NSExpressionDescription()
+        sumExpressionDescription.name = "totalCount"
+        sumExpressionDescription.expression = sumExpression
+        sumExpressionDescription.expressionResultType = .integer64AttributeType
+
+        request.propertiesToFetch = [sumExpressionDescription]
+        request.resultType = .dictionaryResultType
+
+        let results = try context.fetch(request) as? [[String: Any]]
+        return (results?.first?["totalCount"] as? Int64) ?? 0
```

Comment:
I have a question here about how we would hit the `?? 0`.

That happens becaue we do not have records? Or does it happen because of an error of the schema and it is something we do not expect? If it is the latter I would be inclined to throw so we can measure that accordingly wiht the `failedToFetchPrivacyStatsTotalCount` error.

Reply:
This shouldn’t actually happen:
* When the fetch fails, it would throw an error. When it doesn’t fail, it returns `[[String: Any]]` object unless there’s a bug in Core Data :)
* Then, in line 126 we’re traversing through the known JSON structure and it shouldn’t ever be different than what’s here.

We use the same approach in PrivacyStatsUtils, BookmarkUtils and BookmarkCoreDataImporter so I assumed it’s fine as the likelihood of this becoming broken without a huge bug by Apple is really minimal.